### ISSUE TITLE
Why not calcite?

### ISSUE THREAD

**Issue by dohongdayi** [ballista-compute/ballista#331]
May I ask, why not using calcite for SQL planning instead of reinvention?

Thanks

**Comment by zhangchunzhong** [comment]
calcite is in Java. This project is to try rust for big data.

**Comment by dohongdayi** [comment]
What my question was, why not the following modules, which are doing SQL planning in Kotlin, consider calcite:

ballista/jvm/logical-plan
ballista/jvm/optimizer
ballista/jvm/physical-plan
ballista/jvm/QueryPlanner
ballista/jvm/sql

Thanks in advance

**Comment by andygrove** [comment]
Hi @dohongdayi That is a good question. There were a couple of reasons. One is that I really wanted to build this to be Arrow-native from the ground up. Another is that I wrote this SQL parser and planner while writing a book on the subject. Calcite is a great project but I did struggle to find detailed and comprehensive documentation and that made it hard to learn how to really use it effectively and contribute.
